jaxrs-spec/src/main/asciidoc/chapters/providers/_providers.adoc

```diff
@@ -20,6 +20,15 @@ and MAY be annotated with `@Provider` for automatic discovery purposes;
 the integration of pre-packaged providers into the JAX-RS runtime is
 implementation dependent.
 
+[[extensions]]
+Providers MAY be annotated with `@Extension` to declare it as an
+extension of the runtime.
+
+Compatible products MUST automatically register a class in
+`jakarta.ws.rs.core.Configuration` runtime configurations if the
+class is a _provider class_ for the extension _service type_ according
+to the rules set out in the description of `ServiceLoader`.
```

Comment:
I'm not sure if I fully understand how this should be used in practice. If the developer creates something like this:

```java
public class MyFilter implements ContainerRequestFilter, Extension {
   // ...
}
```

Which of these files should the developer use for automatic registration?

  1. `META-INF/services/jakarta.ws.rs.ext.Extension`
  2. `META-INF/services/jakarta.ws.rs.container.ContainerRequestFilter`

I guess it should be (1)?

Reply:
Good catch! In fact, I missed to emphasize the `Extension` word in line 29. I will fix that to make it clear that actually proposal 1 (`META-INF/services/jakarta.ws.rs.ext.Extension`) is the one to use.

Reply:
Done :-)